assets/js/atomic/blocks/product-elements/average-rating/block.tsx

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { useProductDataContext } from '@woocommerce/shared-context';
+import { useStyleProps } from '@woocommerce/base-hooks';
+import { __ } from '@wordpress/i18n';
+
+type ProductAverageRatingProps = {
+	className?: string;
+	textAlign?: string;
+};
+
+export const Block = (
+	props: ProductAverageRatingProps
+): JSX.Element | undefined => {
+	const { textAlign } = props;
+	const styleProps = useStyleProps( props );
+	const { product } = useProductDataContext();
+
+	const className = classnames(
+		styleProps.className,
+		'wc-block-components-product-average-rating',
+		{
+			[ `has-text-align-${ textAlign }` ]: textAlign,
+		}
+	);
+
+	return (
+		<div className={ className } style={ styleProps.style }>
+			{ Number( product.average_rating ) > 0
+				? product.average_rating
+				: __( 'No ratings', 'woo-gutenberg-products-block' ) }
+		</div>
+	);
+};
+
+export default Block;
```

Comment:
Is there any reason we are not using `withProductDataContext` HOC here? Which is used in other product elements, for example:
https://github.com/woocommerce/woocommerce-blocks/blob/a5c203b159622ccd4b88b3d3082f0717255e0d46/assets/js/atomic/blocks/product-elements/rating-counter/block.tsx#L88